src/binary_to_text/binary_encoder.ml

```diff
@@ -0,0 +1,713 @@
+(* SPDX-License-Identifier: AGPL-3.0-or-later *)
+(* Copyright © 2021-2024 OCamlPro *)
+(* Written by the Owi programmers *)
+
+open Binary
+open Syntax
+open Types
+
+let end_opcode = '\x0B'
+
+let magic = "\x00\x61\x73\x6d"
+
+let version = "\x01\x00\x00\x00"
+
+let add_char c buf = Buffer.add_char buf c
+
+let add_string str buf = Buffer.add_string buf str
+
+(* add byte from int (ascii code) *)
+let add_byte i =
+  let c = Char.chr (i land 0xff) in
+  add_char c
+
+(* add 2 bytes (16 bits) from int *)
+let add_bytes_2 i buf =
+  add_byte (i land 0xff) buf;
+  add_byte (i lsr 8) buf
+
+(* add 4 bytes (32 bits) from int32 *)
+let add_bytes_4 i buf =
+  add_bytes_2 (Int32.to_int (Int32.logand i 0xffffl)) buf;
+  add_bytes_2 (Int32.to_int (Int32.shift_right_logical i 16)) buf
+
+(* add 8 bytes (64 bits) from int64 *)
+let add_bytes_8 i buf =
+  add_bytes_4 (Int64.to_int32 (Int64.logand i 0xffffffffL)) buf;
+  add_bytes_4 (Int64.to_int32 (Int64.shift_right i 32)) buf
+
+let rec write_U64 i buf =
+  let b = Int64.to_int (Int64.logand i 0x7fL) in
+  if 0L <= i && i < 128L then add_byte b buf
+  else (
+    add_byte (b lor 0x80) buf;
+    write_U64 (Int64.shift_right_logical i 7) buf )
```

Comment:
Could you use:

```ocaml
else begin
  ...
end
```

instead of parenthesis ? It's more common and thus easier to read.